R/cache-methods.R

```diff
@@ -0,0 +1,203 @@
+# cache-wide stuff
+
+#' @include cache.R
+
+save_cache <- function(cache, ...) {
+  UseMethod("save_cache", cache)
+}
+
+save_cache.environment <- function(cache, file, path = NULL) {
+  if (is.null(path)) {
+    # Only retrieve settings if `path` isn't specified.
+    # Otherwise, other settings may inadvertently be loaded, such as exclusions.
+    read_settings(file)
+    path <- settings$cache_directory
+  }
+
+  if (!file.exists(path)) {
+    dir.create(path, recursive = TRUE)
+  }
+
+  save(
+    file = get_cache_file_path(file, path),
```

Comment:
There should be a NULL check for path here.

Reply:
That should be handled earlier. If settings$cache_directory is null, then there should be an early return

Reply:
Yeah, adding the check in the `if (is.null(path))` branch after l.14 is much better.